test/e2e/kappcontroller/fetch_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package kappcontroller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/test/e2e"
+	"sigs.k8s.io/yaml"
+)
+
+func Test_App_FetchDirname_Single(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+	name := "test-app-fetch-dirname-single"
+
+	appSingleFetchYAML := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: test-app-fetch-dirname-single
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      destDirName: foo
+      paths:
+        file.csv: |
+            Year,Make,Model,Description,Price
+            1997,Ford,E350,"ac, abs, moon",3000.00
+  template:
+  - ytt:
+      paths:
+      - foo
+      inline:
+        paths:
+          file.yml: |
+            #@ load("@ytt:data", "data")
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+              name: configmap
+            data:
+              value: #@ data.read("file.csv")
+  deploy:
+  - kapp: {}
+` + sas.ForNamespaceYAML()
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy", func() {
+		// if template stage succeeds, assume test pass
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			e2e.RunOpts{StdinReader: strings.NewReader(appSingleFetchYAML)})
+	})
+
+	out := kapp.Run([]string{"inspect", "-a", name, "--raw", "--tty=false", "--filter-kind=App"})
+
+	var cr v1alpha1.App
+	err := yaml.Unmarshal([]byte(out), &cr)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal: %s", err)
+	}
+
+	if !strings.Contains(cr.Status.Fetch.Stdout, "path: foo") {
```

Comment:
nit  - let's use stretchr/testify and to `assert.Contains(t, cr.status.Fetch.Stdout, "path: foo", "Expected destDirName to be used by vendir")`